src/OutputBufferStream.php

```diff
@@ -0,0 +1,257 @@
+<?php
+
+namespace Jasny\HttpMessage;
+
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * An instance wraps a PHP stream and can be used for a PSR-7 implementation.
+ * This interface provides a wrapper around the most common operations, including
+ * serialization of the entire stream to a string.
+ */
+class OutputBufferStream implements StreamInterface
+{
+    /**
+     * @var resource 
+     */
+    protected $handle;
+    
```

Comment:
The most important methods are missing:

* `useGlobally()` should set the handle from `php://temp` to `php://output`. But before doing so, it should clear the output buffer (`ob_clean()`) and copy all the contents of `php://temp` to `php://output`.
* `useLocally()` should set the handle from `php://output` to `php://temp`. But before doing so, copy the whole output buffer in `php://temp` and than do `ob_clean()`.

Calling `useGlobally()` when the handle is already `php://output` should do nothing. Same with `useLocally()` when it's not `php://output`.

Add `@ignore` to these methods, as they should only be used by the `Response` object.